projects/storefrontlib/layout/a11y/keyboard-focus/services/select-focus.util.ts

```diff
@@ -26,19 +26,19 @@ export class SelectFocusUtility {
   protected focusableSelectorSuffix =
     ':not([disabled]):not([hidden]):not([aria-hidden])';
 
-  query(host: HTMLElement, selector: string): HTMLElement[] {
+  query(host: HTMLElement | null | undefined, selector: string): HTMLElement[] {
     if (!selector || selector === '') {
       return [];
     }
     return Array.from(
-      host.querySelectorAll(selector) as NodeListOf<HTMLElement>
+      host?.querySelectorAll(selector) as NodeListOf<HTMLElement>
     );
   }
 
   findFirstFocusable(
-    host: HTMLElement,
+    host: HTMLElement | null | undefined,
```

Comment:
I removed undefined from the type, and it still works. Why is it here?

Reply:
same host

Reply:
makes sense! it ties to this here https://github.com/SAP/spartacus/pull/15658#discussion_r870762291 since we're saying to put | undefined instead